docs/guides/model-cache.mdx

```diff
@@ -0,0 +1,97 @@
+- `ignore_patterns`: Conversely, you can also denote file patterns to ignore, hence streamlining the caching process.
+
+Here is an example of a well written `hf_cache` for Stable Diffusion XL. Note how it only pulls the model weights that it needs using `allow_patterns`.
+
+```yaml config.yml
+...
+hf_cache:
+  - repo_id: madebyollin/sdxl-vae-fp16-fix
+    allow_patterns:
+      - config.json
+      - diffusion_pytorch_model.safetensors
+  - repo_id: stabilityai/stable-diffusion-xl-base-1.0
+    allow_patterns:
+      - "*.json"
+      - "*.fp16.safetensors"
+      - sd_xl_base_1.0.safetensors
+  - repo_id: stabilityai/stable-diffusion-xl-refiner-1.0
+    allow_patterns:
+      - "*.json"
+      - "*.fp16.safetensors"
+      - sd_xl_refiner_1.0.safetensors
+...
+```
+
+Many Hugging Face repos have model weights in different formats (`.bin`, `.safetensors`, `.h5`, `.msgpack`, etc.). You only need one of these most of the time. To minimize cold starts, ensure that you only cache the weights you need.
+
+There are also some additional steps depending on the cloud bucket you want to query.
+
+### Hugging Face 🤗
+For any public Hugging Face repo, you don't need to do anything else. Adding the `hf_cache` key with an appropriate `repo_id` should be enough.
+
+However, if you want to deploy a model from a gated repo like [Llama 2](https://huggingface.co/meta-llama/Llama-2-70b-chat-hf) to Baseten, there's a few steps you need to take:
+<Steps>
+  <Step title="Get Hugging Face API Key">
+    [Grab an API key](https://huggingface.co/settings/tokens) from Hugging Face with `read` access. Make sure you have access to the model you want to serve.
+  </Step>
+  <Step title="Add it to Baseten Secrets Manager">
+    Paste your API key in your [secrets manager in Baseten](https://app.baseten.co/settings/secrets) under the key `hf_access_token`.
+  </Step>
+  <Step title="Update Config">
+    In your Truss's `config.yml`, add the following code:
+
+```yaml config.yml
+...
+secrets:
+    hf_access_token: null
+...
+```
+
+Make sure that the key `secrets` only shows up once in your `config.yml`. If you are testing Truss builds locally, you can paste your actual `hf_access_token` here. 
+  </Step>
+</Steps>
+
+If you run into any issues, run through all the steps above again and make sure you did not misspell the name of the repo or paste an incorrect API key.
+
+<Tip>
+Weights will be cached in the default Hugging Face cache directory, `~/.cache/huggingface/hub/models--{your_model_name}/`. You can change this directory by setting the `HF_HOME` or `HUGGINGFACE_HUB_CACHE` environment variable in your `config.yml`. 
+
+[Read more here](https://huggingface.co/docs/huggingface_hub/guides/manage-cache).
+</Tip>
+
+### Google Cloud Storage
+Similar to above, you do not need to do anything else for public GCS buckets.
+
+For a private GCS bucket, first export your service account key. Rename it to be `service_account.json` and add it to the `data` directory of your Truss.
```

Comment:
I think it would be good to flesh this out to, so the directory structure of your truss should be:

my-truss
  |--model
  |     └──model.py
  |--data
  |.     └── service_account.json
  ...
  
  and then maybe mention to not check it into git